src/main/java/com/diffplug/gradle/eclipse/apt/GenerateEclipseFactorypath.java

```diff
@@ -38,10 +38,10 @@ protected void configure(Factorypath factorypath) {
     EclipseFactorypath factorypathModel = getFactorypath();
     factorypathModel.getFile().getBeforeMerged().execute(factorypath);
     Set<File> entries = new LinkedHashSet<>();
-    for (Configuration configuration : factorypathModel.getPlusConfigurations()) {
+    for (FileCollection configuration : factorypathModel.getPlusConfigurations()) {
       entries.addAll(configuration.getFiles());
     }
-    for (Configuration configuration : factorypathModel.getMinusConfigurations()) {
+    for (FileCollection configuration : factorypathModel.getMinusConfigurations()) {
```

Comment:
When the configuration cache is enabled, Gradle will automatically convert disallowed types into cache-friendly types that are used at execution. This means that when storing configurations in tasks, at execution, you can only refer to them as the `FileCollection` type.